### PR TITLE
Add report based on cars with orders

### PR DESCRIPTION
**Pull request by NastyaBondarenko**

### CODE REVIEW

**Review by Eugene7011 — APPROVED** [review]

File: goods-partner-backend/src/test/java/com/goodspartner/report/CarsReportGeneratorITest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        Consumer<ReportResult> reportResultConsumer = r -> {
+            File destinationFile = new File(reportsDestination, r.name());
+            writeReportToFile(r, destinationFile);
+
+            assertTrue(destinationFile.exists());
+            assertTrue(destinationFile.length() > 0);
+        };
+
+        carsReportGenerator.generateReport(date, reportResultConsumer);
```
Comment (BogdanParashchak) [inline goods-partner-backend/src/test/java/com/goodspartner/report/CarsReportGeneratorITest.java:43]: mock service which usee google api key

**Review by BogdanParashchak — CHANGES_REQUESTED** [review]
build fails
for test mock service which uses google api key

File: goods-partner-backend/src/test/java/com/goodspartner/report/CarsReportGeneratorITest.java (hunk 1 above)
Comment (NastyaBondarenko) [reply goods-partner-backend/src/test/java/com/goodspartner/report/CarsReportGeneratorITest.java:158]: done

**Review by BogdanParashchak — APPROVED** [review]

**Review by BogdanParashchak — APPROVED** [review]

File: goods-partner-backend/src/main/java/com/goodspartner/report/CarsReportGenerator.java [hunk 2]
```diff
@@ -117,15 +118,17 @@ private void addRowsForProductsAndFill(XSSFSheet sheet, RoutesCalculation routes
...
-
-        for (int i = 4; i < 4 + (ordersQuantity - 1) * 2; i++) {
-            Row sourceWeightRow = sheet.getRow(4 + (ordersQuantity - 1) * 2);
-            Row newRow = sheet.createRow(i);
-            if (i % 2 == 0) {
-                createCell(sourceWeightRow, newRow);
-            } else {
-                createCell(sourceProductRow, newRow);
+        if (ordersQuantity > 1) {
+            sheet.shiftRows(4, 5, (ordersQuantity - 1) * 2);
```
Comment (Rooman) [inline goods-partner-backend/src/main/java/com/goodspartner/report/CarsReportGenerator.java:122]: Magic numbers. 
Can we use some variables instead like
HEADER_INDEX = 4;

Comment (NastyaBondarenko) [reply goods-partner-backend/src/main/java/com/goodspartner/report/CarsReportGenerator.java]: added

**Review by Rooman — APPROVED** [review]

**Review by Rooman — APPROVED** [review]

**Review by BogdanParashchak — APPROVED** [review]

**Pull request merged**